### PR TITLE
Generalize post-processing code 

### PR DESCRIPTION
**Pull request by SSaishruthi**
This PR,

- updates post-processing code to not use imagenet specific utils.
- Dockerfile, config and md5sum files are updated along.

This is needed for training purpose.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,14 +21,15 @@ ARG model_file=assets.tar.gz
 
 WORKDIR /workspace
 RUN wget -nv --show-progress --progress=bar:force:noscroll ${model_bucket}/${model_file} --output-document=assets/${model_file} && \
-  tar -x -C assets/ -f assets/${model_file} -v && rm assets/${model_file} && \
-  mkdir -p ~/.keras/models && mv assets/imagenet_class_index.json ~/.keras/models/imagenet_class_index.json
+  tar -x -C assets/ -f assets/${model_file} -v && rm assets/${model_file}
+  #mkdir -p ~/.keras/models && mv assets/imagenet_class_index.json ~/.keras/models/imagenet_class_index.json
```
Comment (bdwyer2) [inline Dockerfile]: Can we just delete this line if we aren't using it?

File: Dockerfile [hunk 2]
```diff
@@ -21,14 +21,15 @@ ARG model_file=assets.tar.gz
...
 
 COPY requirements.txt /workspace
 RUN pip install -r requirements.txt
 
 COPY . /workspace
 
+RUN cd assets/ && ls
```
Comment (bdwyer2) [inline Dockerfile]: Is this needed?

Comment (SSaishruthi) [reply Dockerfile]: Removed

File: core/model.py [hunk 3]
```diff
@@ -70,7 +71,16 @@ def _pre_process(self, image):
         return imagenet_utils.preprocess_input(image, mode=self.MODEL_MODE)
 
     def _post_process(self, preds):
-        return imagenet_utils.decode_predictions(preds)[0]
+        preds_sorted_index = preds[0].argsort()[-5:][::-1]
+        top_preds_prob = preds[0][preds_sorted_index]
+        data = json.loads(open(DEFAULT_MODEL_PATH + 'imagenet_class_index.json').read())
```
Comment (MLnick) [inline core/model.py]: This should rather be loaded during `__init__` once

Also use a `with open(...` to properly handle file closing

File: core/model.py [hunk 4]
```diff
@@ -70,7 +71,16 @@ def _pre_process(self, image):
...
+        result = []
+        for i in range(len(preds_sorted_index)):
+            result.append([data[str(preds_sorted_index[i])][0],
+                           data[str(preds_sorted_index[i])][1],
+                           top_preds_prob[i]
+                           ])
```
Comment (xuhdev) [inline core/model.py]: This can be probably done in a clean list comprehension

File: core/model.py [hunk 5]
```diff
@@ -49,7 +50,7 @@ class ModelWrapper(MAXModelWrapper):
     def __init__(self, path=DEFAULT_MODEL_PATH):
         logger.info('Loading model from: {}...'.format(path))
         clear_session()
-        self.model = models.load_model(path)
+        self.model = models.load_model(path + 'resnet50.h5')
```
Comment (xuhdev) [inline core/model.py]: neat: `os.path.join(path, 'resnet50.h5')` (this will handle the case in which users do not add a slash to the end of their path.)

**Comment by SSaishruthi** [comment]
Is this approach ok? If yes, I will merge and create PR for the trainable model.

**Comment by SSaishruthi** [comment]
@MLnick Is this change ok to merge?

**Pull request merged**